src/components/organisms/product-variant-tree/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { Product } from "@medusajs/medusa"
+import React from "react"
+import DollarSignIcon from "../../fundamentals/icons/dollar-sign-icon"
+import TrashIcon from "../../fundamentals/icons/trash-icon"
+import { CollabsibleTree } from "../../molecules/collabsible-tree"
+
+type LeafProps = {
+  id: string
+  title: string
+  sku?: string
+  prices: {
+    id: string
+    currency_code: string
+    amount: number
+  }[]
+}
+
+type ProductVariantTreeProps = {
+  product: Pick<Product, "title" | "id" | "thumbnail"> & {
+    variants: LeafProps[]
+  }
+}
+
+const ProductVariantTree: React.FC<ProductVariantTreeProps> = ({ product }) => {
+  return (
+    <CollabsibleTree>
+      <CollabsibleTree.Parent
+        actions={[
+          {
+            label: "Edit",
+            onClick: () => console.log("Edit prices"), // temp - should open edit prices overrides modal with "Apply on all variants" checked
+            icon: <DollarSignIcon />,
+          },
+          {
+            label: "Delete",
+            onClick: () => console.log("Remove prices for variant"), // temp - should delete all money amounts for variants from product in price list
+            icon: <TrashIcon />,
+            variant: "danger",
+          },
+        ]}
+      >
+        <div>
+          <img src={product.thumbnail} className="w-4 h-5 rounded-base" />
+        </div>
+        <span className="inter-small-semibold">{product.title}</span>
+      </CollabsibleTree.Parent>
+      {product.variants.map((variant) => 
+        <ProductVariantLeaf key={variant.id} {...variant} />
+      )}
```

Comment:
This breaks the component because in `CollapsibleTree`, we are iterating over these leaves and checking if `child.props.__TYPE === "CollapsibleTreeLeaf"` which will always be false because what `CollapsibleTree` sees is `ProductVariantLeaf` and not `CollapsibleTree.Leaf`